### DISCUSSION TITLE
can i reuse my existing eventLoopGroup on my netty-base server

### DISCUSSION THREAD

**Discussion by AlaskaBay** [redis/lettuce#2189] [Q&A] [answered]
i'm developing a http server use netty, bootstrap code like this:

>             bossGroup = Epoll.isAvailable() ? new EpollEventLoopGroup(1) : new NioEventLoopGroup(1);
>             workerGroup = Epoll.isAvailable() ? new EpollEventLoopGroup(3) : new NioEventLoopGroup(3);
>             buzThreadPool = new DefaultEventExecutorGroup(WORKER_NUMBER,
>                     new BasicThreadFactory.Builder().namingPattern("buz-thread-%d").daemon(false).priority(5).build(), 10000,
>                     (r, executor) -> {
>                         LOGGER.log(F3.level, "buz threadPool full! run in caller.");
>                         if (!executor.isShutdown()) {
>                             r.run();
>                         }
>                     });

i want use lettuce to construct fully asynchronous app (since Jedis is blocking io and need too many redis connection).
question: can i reuse existing eventLoopGroup in my program? i see closed issue:[https://github.com/lettuce-io/lettuce-core/issues/283](url) ,it's seems to set eventLoopGroupProvider in ClientResources, but i find source code in DefaultClientResources.java (lettuce-core:5.3.5.RELEASE)
### 

`

>         if (builder.eventLoopGroupProvider == null) {
>             int ioThreadPoolSize = builder.ioThreadPoolSize;
> 
>             if (ioThreadPoolSize < MIN_IO_THREADS) {
>                 logger.info("ioThreadPoolSize is less than {} ({}), setting to: {}", MIN_IO_THREADS, ioThreadPoolSize,
>                         MIN_IO_THREADS);
>                 ioThreadPoolSize = MIN_IO_THREADS;
>             }
> 
>             this.sharedEventLoopGroupProvider = false;
>             this.eventLoopGroupProvider = new DefaultEventLoopGroupProvider(ioThreadPoolSize);
> 
>         } else {
>             this.sharedEventLoopGroupProvider = builder.sharedEventLoopGroupProvider;
>             this.eventLoopGroupProvider = builder.eventLoopGroupProvider;
>         }
> 
>         if (builder.eventExecutorGroup == null) {
>             int computationThreadPoolSize = builder.computationThreadPoolSize;
>             if (computationThreadPoolSize < MIN_COMPUTATION_THREADS) {
> 
>                 logger.info("computationThreadPoolSize is less than {} ({}), setting to: {}", MIN_COMPUTATION_THREADS,
>                         computationThreadPoolSize, MIN_COMPUTATION_THREADS);
>                 computationThreadPoolSize = MIN_COMPUTATION_THREADS;
>             }
> 
>             eventExecutorGroup = DefaultEventLoopGroupProvider.createEventLoopGroup(DefaultEventExecutorGroup.class,
>                     computationThreadPoolSize);
>             sharedEventExecutor = false;
>         } else {
>             sharedEventExecutor = builder.sharedEventExecutor;
>             eventExecutorGroup = builder.eventExecutorGroup;
>         }

`

 i'm confusing now...

**Comment by mp911de** [comment, marked as answer]
Yes, you can reuse eventloops from your netty server. If you implement a `EventLoopGroupProvider`, you can provide an eventloop object that matches the requested channel type (epoll, kqueue, nio, …).
